sbi/inference/trainers/vfpe/base_vf_inference.py

```diff
@@ -632,3 +659,798 @@ def _loss(
+        **kwargs,
+    ):
+        """Base class for masked vector field inference methods. It is
+        used for Simformer.
+
+        NOTE: Masked Vector field inference does not support multi-round inference
+        with flexible proposals yet.
+
+        Args:
+            prior: Prior distribution. Its primary use is for rejecting samples that
+                fall outside its defined support. For the core inference process,
+                this prior is ignored, as the actual "prior" over which the diffusion
+                model operates is standard Gaussian noise.
+            mvf_estimator_builder: Neural network architecture for the
+                masked vector field estimator. Can be a string (e.g. 'simformer')
+                or a callable that implements the `MaskedVectorFieldEstimatorBuilder`
+                protocol with `__call__` that receives `inputs`, `condition_mask`,
+                and `edge_mask` and returns a `MaskedConditionalVectorFieldEstimator`.
+            device: Device to run the training on.
+            logging_level: Logging level for the training. Can be an integer or a
+                string.
+            summary_writer: Tensorboard summary writer.
+            show_progress_bars: Whether to show progress bars during training.
+            kwargs: Additional keyword arguments passed to the default builder if
+                `vector_field_estimator_builder` is a string.
+        """
+
+        super().__init__(
+            prior=prior,
+            device=device,
+            logging_level=logging_level,
+            summary_writer=summary_writer,
+            show_progress_bars=show_progress_bars,
+        )
+
+        # As detailed in the docstring, `vector_field_estimator` is either a string or
+        # a callable. The function creating the neural network is attached to
+        # `_build_neural_net`. It will be called in the first round and receive
+        # thetas and xs as inputs, so that they can be used for shape inference and
+        # potentially for z-scoring.
+        check_estimator_arg(mvf_estimator_builder)
+        if isinstance(mvf_estimator_builder, str):
+            self._build_neural_net = self._build_default_nn_fn(
+                vector_field_estimator_builder=mvf_estimator_builder,
+                **kwargs,
+            )
+        else:
+            self._build_neural_net = mvf_estimator_builder
+
+        self._proposal_roundwise = []
+
+        self.latent_idx = (
+            torch.as_tensor(latent_idx, dtype=torch.long)
+            if latent_idx is not None
+            else None
+        )
+        self.observed_idx = (
+            torch.as_tensor(observed_idx, dtype=torch.long)
+            if observed_idx is not None
+            else None
+        )
+
+    @abstractmethod
+    def _build_default_nn_fn(self, **kwargs) -> MaskedVectorFieldEstimatorBuilder:
+        pass
+
+    # TODO: Introduced to avoid conflicts, should adjust
```

Comment:
Mhh, I think this has to do with the new PR merged into main which moved building the posterior into "NeuralInference". Basically every posterior now calls this [here](https://github.com/sbi-dev/sbi/blob/5cec1eaf284e1e57376a161e99ccb3b154fa7c89/sbi/inference/trainers/base.py#L397).

What you would need to add here is to use the conditional estimator wrapper.